src/main/java/ch/njol/skript/classes/registry/RegistryParser.java

```diff
@@ -0,0 +1,155 @@
+		for (R registryObject : registry) {
+			NamespacedKey namespacedKey = registryObject.getKey();
+			String namespace = namespacedKey.getNamespace();
+			String key = namespacedKey.getKey();
+			String keyWithSpaces = key.replace("_", " ");
+			String languageKey = languageNode + "." + key;
+
+			// Put the full namespaced key as a pattern
+			parseMap.put(namespacedKey.toString(), registryObject);
+
+			// If the object is a vanilla Minecraft object, we'll add the key with spaces as a pattern
+			if (namespace.equalsIgnoreCase(NamespacedKey.MINECRAFT)) {
+				parseMap.put(keyWithSpaces, registryObject);
+			}
+
+			String[] options = Language.getList(languageKey);
+			// Missing/Custom registry objects
+			if (options.length == 1 && options[0].equals(languageKey.toLowerCase(Locale.ENGLISH))) {
+				if (namespace.equalsIgnoreCase(NamespacedKey.MINECRAFT)) {
+					// If the object is a vanilla Minecraft object, we'll use the key with spaces as a name
+					names.put(registryObject, keyWithSpaces);
+				} else {
+					// If the object is a custom object, we'll use the full namespaced key as a name
+					names.put(registryObject, namespacedKey.toString());
+				}
+			} else {
+				for (String option : options) {
+					option = option.toLowerCase(Locale.ENGLISH);
+
+					// Isolate the gender if one is present
+					NonNullPair<String, Integer> strippedOption = Noun.stripGender(option, languageKey);
+					String first = strippedOption.getFirst();
+					Integer second = strippedOption.getSecond();
+
+					// Add to name map if needed
+					names.putIfAbsent(registryObject, first);
+
+					parseMap.put(first, registryObject);
+					if (second != -1) { // There is a gender present
+						parseMap.put(Noun.getArticleWithSpace(second, Language.F_INDEFINITE_ARTICLE) + first, registryObject);
+					}
+				}
+			}
+		}
+	}
+
+	/**
+	 * This method attempts to match the string input against one of the string representations of the registry.
+	 *
+	 * @param input a string to attempt to match against one in the registry.
+	 * @param context of parsing, may not be null
+	 * @return The registry object matching the input, or null if no match could be made.
+	 */
+	@Override
+	public @Nullable R parse(String input, @NotNull ParseContext context) {
+		return parseMap.get(input.toLowerCase(Locale.ENGLISH));
+	}
+
+	/**
+	 * This method returns the string representation of a registry.
+	 *
+	 * @param object The object to represent as a string.
+	 * @param flags  not currently used
+	 * @return A string representation of the registry object.
+	 */
+	@Override
+	public @NotNull String toString(R object, int flags) {
+		return names.get(object);
+	}
+
+	/**
+	 * Returns a registry object's string representation in a variable name.
+	 *
+	 * @param object Object to represent in a variable name.
+	 * @return The given object's representation in a variable name.
+	 */
+	@Override
+	public @NotNull String toVariableNameString(R object) {
+		return toString(object, 0);
```

Comment:
this might create a variable like `{_flooby:floppity}` no? that may result in a questionable name:
https://github.com/SkriptLang/Skript/blob/0355df1a07c027ca2fb5355eeaa57378d524f813/src/main/java/ch/njol/skript/lang/Variable.java#L165-L169
however, i am not sure how we might want the name to display in a variable

Reply:
from what I know, this should be fine.
Small example:
```vb
on load:
	set {_var::%minecraft name of stone sword%} to 1 of stone sword
	send indexes of {_var::*}
```
out: `minecraft:stone_sword`

Reply:
hmm since it's not recommended though do you think we should do it? maybe it might be better to replace it with an underscore but im not sure

Reply:
I think it's fine because it's what you'd expect it to be.

Reply:
this already happens with entities in indices anyway: entity:82892e92e9...

Reply:
sounds good